### DISCUSSION TITLE
Returning an action from another action

### DISCUSSION THREAD

**Discussion by shannoga** [pointfreeco/swift-composable-architecture#2422] [Q&A]
Assuming that I have to return an action from another action (We know we should try to avoid chaining actions, but in case that this is logical).

Is it valid / good practice to simply -
` return EffectTask(value: .anyAction)`

Since the reduce function is expecting an EffectTask to be returned, is there a reason to avoid that?

Thanks!

**Comment by maksimn** [comment]
Hello, @shannoga 
It is better to do this:
```
return .send(.someAction)
```

**Comment by mbrandonw** [comment]
Hi @shannoga, @maksimn is correct that `.send(…)` is a shorter way of constructing an effect that simply emits another action, but I do want to make sure you are aware of [this documentation](https://pointfreeco.github.io/swift-composable-architecture/main/documentation/composablearchitecture/performance#Sharing-logic-with-actions) about the downsides to overusing this pattern.
